packages/alfa-aria/src/feature.ts

```diff
@@ -167,13 +167,27 @@ const nameFromChild = (predicate: Predicate<Element>) => (
   return Branched.of(None);
 };
 
+const ids = Cache.empty<Node, Map<string, Element>>();
+
+const labels = Cache.empty<Node, Sequence<Element>>();
+
 const nameFromLabel = (element: Element, device: Device, state: Name.State) => {
   const root = element.root();
 
+  const elements = root.inclusiveDescendants().filter(isElement);
```

Comment:
I think this should either be cached (like `ids` and `labels`) or rather just skip the computation if we `ids.has(root) && labels.has(root)` since it seems that `elements` is only used in these cache misses.

Reply:
It's a lazy sequence so without enumerating it nothing is actually computed.